dev/tasks/r/azure.linux.yml

```diff
@@ -50,6 +50,7 @@ jobs:
           export ARROW_R_DEV={{ not_cran|default("TRUE") }}
           # Note that ci/scripts/r_test.sh sets NOT_CRAN=true if ARROW_R_DEV=TRUE
           docker-compose run \
+            {{ flags|default("") }} \
```

Comment:
This is a bit of 🚲 🏠, but should we call this something like `r_additional_flags`? To make clear where they are headed?

Reply:
We should rename it (I noted similar on Nic's PR), but it's not `r_additional_flags`, it's `docker_args` or something

Reply:
I like expanding it to be useable elsewhere, though if it's `docker_args` we should add it to the other tasks then, yeah?

Reply:
Yeah there a bit of cleanup/refactoring to do (https://github.com/apache/arrow/pull/10710#discussion_r681756880)

Reply:
Ah, great — reviewing that is next on my to-do list after all of this LTO shenanigans

Reply:
ARROW-13586